### PR TITLE
[WIP]: refactor and get build working

### PR DESCRIPTION
**Pull request by DSchau**
This PR _almost_ fixes everything. The central issues were:

- gatsby-plugin-feed was configured incorrectly (you can serialize multiple feeds with one instance of the plugin, rather than multiple)
- gatsby-mdx seems to have an issue with its html resolver currently
    - You can validate with the following query (this breaks the local development server)
    ```graphql
    {
      mdx(fields:{
        slug:{
          eq:"/blog/compound-components-with-react-hooks"
        }
      }) {
        html
        frontmatter {
          title
        }
      }
    }
    ```

    ```
    ReferenceError: document is not defined
        at createCache (evalmachine.<anonymous>:3662:38)
        at Module../node_modules/@emotion/core/dist/core.browser.esm.js (evalmachine.<anonymous>:3807:149)
        at __webpack_require__ (evalmachine.<anonymous>:724:30)
        at fn (evalmachine.<anonymous>:101:20)
        at Module.<anonymous> (evalmachine.<anonymous>:27685:117)
        at Module../node_modules/gatsby-mdx/utils/wrap-root-render-html-entry.js (evalmachine.<anonymous>:27732:30)
        at __webpack_require__ (evalmachine.<anonymous>:724:30)
        at evalmachine.<anonymous>:791:100
        at evalmachine.<anonymous>:794:10
        at Script.runInContext (vm.js:134:20)
    ```

I also made the following tweaks:

- Use `gatsby serve` instead of `serve`
    - Does the same thing, just with some gatsby-oriented tweaks/features
- Co-locate the blog/writing queries
    - It's a little cleaner to just get both data sources at the same time, but then send your reusable blog post helper data rather than having it handle the query
  
This does fix the build (🎉) but unfortunately we need to fix the html resolution in gatsby-mdx before we can get this merged out. For now, I've just removed the `html` field so you can validate that this does, in fact, build out.

I'll work with @chrisbiscardi on identifying the underlying HTML issue.

I'll also plan to work on gatsby-plugin-feed next week. I think it's incredibly confusing, and there is a lot of room for improvement, particularly in that it doesn't appear to be handling GraphQL errors gracefully or displaying an error at all.

Thanks for surfacing these issues!

### CODE REVIEW

**Comment by kentcdodds** [comment]
Wow Dustin, this is fantastic! Thank you so much for your help :100:

File: package.json [hunk 1]
```diff
@@ -70,7 +69,7 @@
   "scripts": {
     "build": "gatsby build",
     "dev": "gatsby develop",
-    "start": "serve public/",
+    "start": "gatsby serve --port 5000",
```
Comment (DSchau) [inline package.json:72]: This is the same port as serve :) Otherwise we use 9000 by default.

File: package.json [hunk 2]
```diff
@@ -21,9 +21,9 @@
     "gatsby": "^2.1.4",
     "gatsby-image": "^2.0.25",
     "gatsby-link": "^2.0.7",
-    "gatsby-mdx": "^0.3.6",
+    "gatsby-mdx": "^0.4.0",
```
Comment (DSchau) [inline package.json:24]: Bumping versions never hurts as a means of ruling things out. gatsby-mdx in particular is under pretty active development.

(Note: no fixes related to this PR though)

**Review by DSchau — COMMENTED** [review]
Left some little comments!

I'll leave this a WIP, but we could progress beyond the blocker as long as we're comfortable with not including the HTML as the `content:encoded` field--or perhaps including something different in its place (e.g. perhaps the raw Markdown content is OK until we resolve the upstream issue).

**Comment by kentcdodds** [comment]
Thanks. I think I'll disable the RSS feed until we can get this fixed. I really appreciate your help!

**Comment by kentcdodds** [comment]
If you could remove the WIP, I'll go ahead and get this merged :)

**Comment by DSchau** [comment]
@kentcdodds done! Feel free to reach out if I can ever help with anything else!

**Comment by kentcdodds** [comment]
Thank you big time!

**Pull request merged**